src/main/java/uk/gov/hmcts/reform/wataskmanagementapi/auth/permission/PermissionEvaluatorService.java

```diff
@@ -118,6 +89,57 @@ private boolean evaluateAccess(Map<String, CamundaVariable> variables,
         return hasAccess;
     }
 
+    private boolean attributesPermissionCheck(Map<String, CamundaVariable> variables,
+                                              Assignment roleAssignment,
+                                              AttributesValueVerifier attributeEvaluatorService) {
+        boolean hasAccess = true;
+        Map<String, String> attributes = roleAssignment.getAttributes();
+        // 3. Conditionally check Jurisdiction matches the one on the task
+        String jurisdictionAttributeValue = attributes.get(RoleAttributeDefinition.JURISDICTION.value());
+        if (hasAccess && jurisdictionAttributeValue != null) {
+
+            hasAccess = attributeEvaluatorService.hasJurisdictionPermission(
+                jurisdictionAttributeValue, variables);
+            log.debug("Jurisdiction permission check {}", hasAccess);
+        }
+        // 4. Conditionally check CaseId matches the one on the task
+        String caseIdAttributeValue = attributes.get(RoleAttributeDefinition.CASE_ID.value());
+        if (hasAccess && caseIdAttributeValue != null) {
+            hasAccess = attributeEvaluatorService.hasCaseIdPermission(caseIdAttributeValue, variables);
+            log.debug("CaseId permission check {}", hasAccess);
+        }
+        // 5. Conditionally check caseTypeId matches the one on the task
+        String caseTypeValue = attributes.get(RoleAttributeDefinition.CASE_TYPE.value());
+        if (hasAccess && caseTypeValue != null) {
+            hasAccess = attributeEvaluatorService.hasCaseTypeIdPermission(caseTypeValue, variables);
+            log.debug("CaseTypeId permission check {}", hasAccess);
+        }
+
+        if (hasAccess) {
+            hasAccess = optionalAttributesCheck(variables, attributeEvaluatorService, attributes);
+        }
+        return hasAccess;
+    }
+
+    private boolean optionalAttributesCheck(Map<String, CamundaVariable> variables,
```

Comment:
I think all the ones in the attributes are optional the only ones mandatory are grant type and security classification according to hld 1.2